### PR TITLE
DOCSP-12627: [MONGOSH] Transaction, Session, and Replica Set Config methods

### PR DESCRIPTION
**Pull request by steveren**
Includes [DOCSP-13309](https://jira.mongodb.org/browse/DOCSP-13309), [DOCSP-13311](https://jira.mongodb.org/browse/DOCSP-13311), and [DOCSP-12627](https://jira.mongodb.org/browse/DOCSP-12627).

[Staged](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-12627/reference/methods)

### CODE REVIEW

File: source/reference/methods.txt [hunk 1]
```diff
@@ -730,6 +730,233 @@ Query Plan Cache Methods
...
+
+   * - :method:`rs.config()`
+
+     - Returns a document that contains the current replica set configuration.
+
+   * - :method:`rs.freeze()`
+
+     - Makes the replica set member that ``mongosh`` is connected to 
+       ineligible to become primary for the specified duration. The 
+       duration must be specified in seconds.
```
Comment (jwilliams-mongo) [inline source/reference/methods.txt:761]: active voice
```suggestion
       ineligible to become primary for the specified duration. You 
       must specify the duration in seconds.
```

File: source/reference/methods.txt [hunk 2]
```diff
@@ -730,6 +730,233 @@ Query Plan Cache Methods
...
+     - Returns the status of the replica set from the oplog data.
+
+   * - :method:`rs.initiate()`
+
+     - Initializes a new replica set.
+
+   * - ``rs.isMaster``
+
+     - Returns the replica set configuration, role, and status for 
+       the ``mongod`` instance that ``mongosh`` is connected to.
```
Comment (jwilliams-mongo) [inline source/reference/methods.txt:774]: adds a link (if this is configured for this repository)
```suggestion
       the |mongod| instance that ``mongosh`` is connected to. 
```

File: source/reference/methods.txt [hunk 3]
```diff
@@ -730,6 +730,233 @@ Query Plan Cache Methods
...
+   * - :method:`rs.initiate()`
+
+     - Initializes a new replica set.
+
+   * - ``rs.isMaster``
+
+     - Returns the replica set configuration, role, and status for 
+       the ``mongod`` instance that ``mongosh`` is connected to. 
+       
+       The ``mongosh`` ``rs.isMaster()`` method wraps the server :dbcommand:`isMaster` command.
```
Comment (jwilliams-mongo) [inline source/reference/methods.txt:776]: `rs.isMaster()` here has parentheses. The reference above does not. Think they should be consistent. Not sure which one is correct.

File: source/reference/methods.txt [hunk 4]
```diff
@@ -730,6 +730,233 @@ Query Plan Cache Methods
...
+
+   * - :method:`db.printReplicationInfo()`
+
+     - Returns the oplog of the replica set member that ``mongosh`` is 
+       connected to. 
+       
+       This is identical to the :method:`rs.printReplicationInfo()` method.
```
Comment (jwilliams-mongo) [inline source/reference/methods.txt:783]: These statements are not used consistently: `This is identical to ...`. If the same applies for methods listed above and below, consider adding these phrases each of those.

An alternative option is to remove these statements. I can see readers viewing them as redundant.

File: source/reference/methods.txt [hunk 5]
```diff
@@ -730,6 +730,233 @@ Query Plan Cache Methods
...
+          }
+
+   * - ``rs.printSecondaryReplicationInfo``
+
+     - Returns the status of the secondary members of the replica set. 
+     
+       This is identical to the ``db.printSecondaryReplicationInfo()`` 
+       method. This method's output is similar to the 
+       :method:`rs.printSlaveReplicationInfo()` method in the legacy 
+       :binary:`~mongo` shell. The :method:`rs.printSlaveReplicationInfo()` method is deprecated in ``mongosh``. Use
```
Comment (jwilliams-mongo) [inline source/reference/methods.txt:826]: if this replacement is configured for this repo.
```suggestion
       |mongo| shell. The :method:`rs.printSlaveReplicationInfo()` method is deprecated in ``mongosh``. Use 
```

**Review by jwilliams-mongo — APPROVED** [review]
LGTM with a few optional suggestions.

File: source/reference/methods.txt (hunk 1 above)
Comment (steveren) [reply source/reference/methods.txt:761]: 👍

File: source/reference/methods.txt (hunk 2 above)
Comment (steveren) [reply source/reference/methods.txt:774]: 👍

File: source/reference/methods.txt (hunk 3 above)
Comment (steveren) [reply source/reference/methods.txt:776]: With parens is correct, I believe.

File: source/reference/methods.txt (hunk 4 above)
Comment (steveren) [reply source/reference/methods.txt:783]: I didn't actually write up these particular methods (just moved them around to put them in alphabetical order) so I'm not sure what the original intent was, but it seems reasonable to remove the "identical" sentences.

File: source/reference/methods.txt (hunk 5 above)
Comment (steveren) [reply source/reference/methods.txt:826]: 👍

**Review by aherlihy — APPROVED** [review]

**Pull request merged**